### PR TITLE
Nerfs the AI in several minor ways.

### PR DESCRIPTION
**Pull request by theOperand**
- Adds a "Camera Failure" event that disables a few cameras.
- Makes tracking take time, increasing with distance from the eye, up to a maximum of four seconds (the AI detector item will light up at the start of the track.)
- AI can no longer track people that are not in its vision, once they are out of vision the tracking stops.
- Makes camera alerts only happen when a camera is reactivated.
- You can now bludgeon cameras to break them. You can just use a wirecutter on cameras to repair them.
- Camera wires are gone. You can open the panel of a camera and then use a wirecutter to disable the camera or a multitool to mess up the focus.

Inspired on [this thread](https://tgstation13.org/phpBB/viewtopic.php?f=10&t=3645).
This adds several minor nerfs intended to reduce meta and make it easier for antags to escape the AI's gaze.

I made this in one and a half hours and can't be assed to test right now. I might remove camera wires completely, but I can't be assed for that right now, so bludgeoning is rather simplistic.

### CODE REVIEW

**Comment by SconesC** [comment]
intense thumb/100 emojis go here

**Comment by CorruptComputer** [comment]
:+1:

**Comment by RemieRichards** [comment]
I was worried about the "time to track" nerf, but as you've set it to a max of 4 seconds, that's acceptable.

**Comment by Miauw62** [comment]
The numbers are entirely debatable, they're just random numbers i slapped on that I thought make sense. Same goes for the camera failure event.

**Comment by RemieRichards** [comment]
I'd be happier with 3 but since everyone's crying for these nerfs, I'll take 4 lol.

**Comment by KorPhaeron** [comment]
Just remove wires. Wire cutter to enable/disable, screwdriver to fuck up the focus, weld a disabled camera to take it off the wall. Nice and simple.

Good stuff in general

**Comment by Kromgar** [comment]
Does this fix emps setting off camera alarms and fire alarms? Maybe send no notice for emp'd fire alarms

**Comment by SconesC** [comment]
>Just remove wires. Wire cutter to enable/disable, screwdriver to fuck up the focus, weld a disabled camera to take it off the wall. Nice and simple.

Would you upgrade cameras just by clicking on them with an analyzer/prox sensor?

File: code/modules/events/camerafailure.dm [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/datum/round_event_control/camera_failure
+	name = "Camera Failure"
+	typepath = /datum/round_event/camera_failure
+	weight = 100
+	max_occurrences = 20
```
Comment (Dorsisdwarf) [inline code/modules/events/camerafailure.dm]: Needs to  be higher if it's a 100-weight chance, IMO

**Comment by CorruptComputer** [comment]
>Would you upgrade cameras just by clicking on them with an analyzer/prox sensor?

I would say remove it from the wall then attach it and put it back on the wall.

**Comment by Dorsisdwarf** [comment]
Screwdrivering a camera should open the panel. 

Then: wirecut to disable power
          multitool to fuck the focus
          analyser/prox/plasma to do the upgrades

Oh, and the time should be from the AI eye, not the AI core

**Comment by Chocobro** [comment]
:+1: v big fan

**Comment by RemieRichards** [comment]
>Oh, and the time should be from the AI eye, not the AI core

@Miauw62 this ^ this ^ this ^

**Comment by Cheridan** [comment]
I still don't like the camera failure event. The AI can be made less dominant without giving it friggin' glaucoma as the round goes on.

**Comment by WJohn** [comment]
Yeah absolutely remove the whole wire mechanic from cameras. It's completely unecessary and makes them way the hell too complicated, when only two of those wires actually matter at all.

**Comment by Miauw62** [comment]
@Cheridan people can fix the cameras, without this there is no way to take suspicion off of disabled cameras.

**Comment by PKPenguin321** [comment]
:+1: next best nerf to removing automatic tracking altogether

**Comment by palpatine213** [comment]
:+1: :x: :100: for using a delay for the tracking rather than removing it altogether.

**Comment by AlexanderUlanH** [comment]
This may be a bigger nerf to the AI's ability to help against wizards more than other antags, just something to consider.

**Comment by Fox-McCloud** [comment]
If I recall, correctly, cameras used to be wirecutters to enable/disable in the past.

**Comment by GunHog** [comment]
While I dislike all of these nerfs, the one I find the worst is the nerf of camera alarms from EMP. Something that powerful needs to have a drawback, or they will need to be made more difficult to obtain. EMP is currently an extremely powerful weapon - It disables radios, saps the charge on energy weapons, wrecks havoc on borgs, shuts down augmented/implanted people, breaks consoles, and completely stomps mechs.

<b>Something this powerful needs a drawback</b>. One or two blasts from your grenade, ion weapon, changeling power, etc turns Sec and Borgs into wet, radio-silent noodles. For something like that to be near zero risk to use (other than not hitting the door you plan to use as an escape route) would be nothing short of overpowered.

**Comment by KorPhaeron** [comment]
@GunHog 

Has a point. EMP weapons are indiscriminate and ruin electronics/cause chaos, they should be loud/bring the AI's attention to you.

**Comment by SconesC** [comment]
I like the AI not getting a ton of alerts.

HOWEVER, I agree EMP shit is too prevalent. I warned about this when Ion Carbines were PR'd. There is a reason that there is a single EMP at roundstart on the station, and it's in the Armory and a large item. Chemists can make EMP with mining materials. That's it.

**Comment by GunHog** [comment]
EMPs are fine as POWERFUL but LOUD. That is what they should be, that is what they should *always* be. That is what the risk is intended to be in using something like that. It will make powerful enemies like borgs, mechs and Security weak and vulnerable, at the cost of attracting more of them too you. To take that away one of the two drawbacks would be grounds for more nerfs. The other drawback is mitigated by knowing how to use your weapon (do not EMP yourself or your escape route).

**Comment by palpatine213** [comment]
Additionally, the laser pointer is able to emp cameras by shining it at them. It's already completely silent and can stun borgs and blind people at range with no indicator. 
With no alarms, a traitor RD or miner, can just quickly shine it at all the cameras, and kill their target, without anyone getting any warning at all.
I don't think the laser even needs to be upgraded to take out cameras.

**Comment by freerealestate** [comment]
An un-upgraded laser has only a very small chance to overload things.

**Comment by PKPenguin321** [comment]
@GunHog 
EMPs do have a drawback, they're rare. You have EMP gear from traitors, the ion rifle in the armory, and MAYBE ion guns from science if they somehow convince security to give them firing pins and aren't incompetent. Cultists never use them and changelings are the only real reliable source of them. That's like, five potential sources, three of which are round-type dependent (and limited to the antagonist) and one of which almost never happens. (Unless EMP chem reactions are still in but I think you need uranium or something, I don't really know goonchem that well)
That means that there is only ONE 100% reliable source of EMPs, and that's the ion rifle in the armory. And that's only really accessible to security and can be spaced or otherwise destroyed relatively easily.

Something that rare doesn't need the drawback of having the AI instantly know there's a traitor and where they are. It makes using EMPs not worth it a very large amount of the time due to abundance of cameras. Considering that most EMPs are from antags, using one is essentially like screaming to the station that THERE IS AN ANTAGONIST RIGHT HERE WHERE THE EMP ALARM WENT OFF.

Remove EMP alarms 2k15

**Comment by ike709** [comment]
>Makes it possible to bludgeon cameras.

Does this set off the camera alarm?

**Comment by Fox-McCloud** [comment]
@ike709 

Not until it's broken, I believe.. It's a Bay feature---IMO, if the wirecutters = killed cameras thing is implemented, it's unnecessary.

**Comment by freerealestate** [comment]
I noticed a couple of bugs:

- Energy swords disable cameras on the first hit despite only doing 30 damage
- Hitting a roundstart disabled camera with a sword will reactivate it somehow
- Cameras become invulnerable to bludgeoning after being reactivated

**Comment by Lumipharon** [comment]
If emp's still notify every silicon that it takes out a camera, then they still serve zero purpose in terms of hiding you from the AI.

Even if they AI can't see you in X room, in still has cameras outside to bolt all the doors until sec comes to shit on you.

The only time emp's become truly OP, is fucking tatorlings.
Spamming emp, while abusing an ebow/armblade/regen/stun break is 10/10 gay.

**Comment by AlexanderUlanH** [comment]
Personally, I like our cameras' wire system.  You can have sabotaged cameras that are slightly effective but not obviously out.
I definitely disagree with removing the alarm wire, since it's trivial to avoid it anyway with a bit of luck or planning.  The only time an AI will get a camera alarm is if the person cutting the cameras hasn't been able to plan ahead by checking the wires or by getting a multitool, which really isn't much effort to do.  I like the option of smashing cameras, but that should also alert the AI, since it's a desperate move.
One thing that would be a good nerf to the AI's cameras is if the coverage were gained with fewer redundant cameras and fewer areas being seen from cameras in harder to reach places, such as heads' and the quartermaster's room.

**Comment by Incomptinence** [comment]
Camera wires are terrible get rid of them.

**Comment by AlexanderUlanH** [comment]
But my focus wire :(

**Comment by Dorsisdwarf** [comment]
Focus could happen if you hit it enough times to damage but not break it

**Comment by RemieRichards** [comment]
@Miauw62 oit butt, Change the distance check to Eye --> person isntead of Core --> person

**Comment by Miauw62** [comment]
EMPs still trigger fire alarms etc.
if you dont remove the very obvious "TRAITOROUS ACTIVITY HERE !!!!!!!!!!!!!" sign EVERY TIME A CAMERA IS DISABLED WITHOUT WIRECUTTING this PR is pointless.

**Comment by xxalpha** [comment]
>Makes tracking take time, increasing with distance from the core (up to a maximum of four seconds)

This should be in Eye instead of Core. It makes more sense and is independent of the map being used.

**Comment by Miauw62** [comment]
im still too lazy to test my own code, unsurprisingly.

**Comment by Stickymayhem** [comment]
I tested it briefly. Tracking mobs that are definitely visible doesn't seem to work.

Also skorvold suggested a cool idea. If you throw any object at a shocked airlock it should spark. Currently there's no way to check without gloves or getting shocked so this is a super neat solution.

**Comment by kevinz000** [comment]
As of right now, there is NO CLICKDELAY on bashing cameras.
That , and even being able to smash cameras, is going to make it easier to smash cameras then smashing lights (Which actually has a chance to shock you if you dont have gloves)

**Comment by Miauw62** [comment]
sure i'll add a clickdelay and look into the tracking not working thing

**Comment by Miauw62** [comment]
this should be good to merge now pls
oh wait clickdelay

**Comment by Miauw62** [comment]
ffs why cant i track anybody

**Comment by SconesC** [comment]
merge these memes

**Comment by freerealestate** [comment]
It still has bugs

**Comment by Miauw62** [comment]
thats the joke

**Comment by Miauw62** [comment]
READY TO MERGE

**Comment by Miauw62** [comment]
you now need to be untrackable for 10 seconds for the track to break, because aran whined at me.

**Comment by Aranclanos** [comment]
also remove the tracking cooldown because its mega frustrating

**Comment by Miauw62** [comment]
more changes ;-;

**Comment by GunHog** [comment]
I still wish you would remove that camera alarm nerf, AoE EMPs should not be stealthy.

**Comment by Miauw62** [comment]
they arent

**Comment by PKPenguin321** [comment]
:pray: :us: :pray:

**Pull request merged**